### DISCUSSION TITLE
Uploading Youtube videos in Create Resources

### DISCUSSION THREAD

**Discussion by siddhi0719** [Sunbird-Ed/Community#643] [General Discussion]
I'm trying to upload videos from YouTube, in Sunbird version 4.6 but I'm having trouble. 
The message that appears is,
"The video you are attempting to upload is not licences under CC-BY. Kindly try a different video."

How can i upload the videos which has Copy Rights licences on it?

**Comment by HarishGangula** [comment]
@siddhi0719  As of now Sunbird supports the creation of the content licenced under CC_BY only for Youtube videos.
cc: @Krishnaj20  @pallakartheekreddy

**Reply by siddhi0719** [reply]
@HarishGangula Is there any options to add license?

**Reply by Dhanashree-Patil95** [reply]
@HarishGangula following is the link of documentation which says Default licenses are available in systems listed below
https://knowlg.sunbird.org/learn/product-and-developer-guide/content-service/content-apis/features#license
can this help to create resource with Standard YouTube License?

**Reply by vinukumar-vs** [reply]
@Dhanashree-Patil95 can you please share the YouTube link that you are trying to upload. Also I would suggest you to share the screenshot where are you trying to upload this YouTube link.

Just to know, are you create new content by using YouTube link or add the YouTube video to the slide(Using Interactive(content) editor).

**Reply by siddhi0719** [reply]
@vinukumar-vs Here is the Youtube [link](https://www.youtube.com/watch?v=Q5JjN2q3XeE) which we are trying to upload and also adding the screenshot 

![Screenshot from 2023-10-25 12-05-34](https://github.com/Sunbird-Ed/Community/assets/147367624/9bfbadb7-0bd0-44ac-9bf4-db6f5da0601e)


We are adding  YouTube video to the slide(Using Interactive(content) editor).

This is how we are uploading:
1. Create -> Resources
2. Add video
3. Upload and Use -> YouTube

**Reply by pallakartheekreddy** [reply]
@siddhi0719 Currently, the editor is making an API call to check the YouTube license, and the default configuration for this API is set to `creativeCommon.` To support different licenses, please update this configuration value to the license you wish to allow.

Reference: https://github.com/Sunbird-Knowlg/sunbird-learning-platform/blob/release-4.6.0/platform-modules/common/src/main/java/org/sunbird/common/util/YouTubeUrlUtil.java#L54

**NOTE:** Please note that the default license value for contents is "CC-BY," which is why only YouTube URLs with this license are allowed to be added within the content. If you decide to update the content license and the video within that content has a different license, please ensure that this is taken into account.

**For example**, if you wish to use the "Standard YouTube License," please update the above configuration value to "youtube." Additionally, when creating content, ensure that the content license is set to "Standard YouTube License" to align with this choice.
@Krishnaj20 @HarishGangula

**Comment by Dhanashree-Patil95** [comment]
@pallakartheekreddy I have few questions regarding above configuration , is there anyone who can help me with this by connecting?

**Reply by pallakartheekreddy** [reply]
@Dhanashree-Patil95 Please post your questions here. If required, we can connect.

**Reply by Dhanashree-Patil95** [reply]
@pallakartheekreddy In our case we need to upload  resources from you tube with "standard you tube license" and also with "creative common" license. Can it be done ?

**Reply by pallakartheekreddy** [reply]
@Dhanashree-Patil95 Yes, The configuration accepts an array of values. Please pass both values in the configuration to allow both licences.

**Reply by Dhanashree-Patil95** [reply]
@pallakartheekreddy Ok .Thank you .

**Reply by pallakartheekreddy** [reply]
@Dhanashree-Patil95 Hope your query is now resolved. Please mark the thread as answered.

**Reply by Dhanashree-Patil95** [reply]
Yes, ok.
